pwnlib/fmtstr.py

```diff
@@ -708,8 +716,52 @@ def make_payload_dollar(data_offset, atoms, numbwritten=0, countersize=4):
 
         # perform write
         if padding:
-            fmt += "%" + str(padding) + "c"
-        fmt += "%" + str(data_offset + idx) + "$" + SPECIFIER[atom.size]
+            # if the padding is less than 3, it is more convenient to write it : [ len("cc") < len("%2c") ] , this could help save some bytes, if it is 3 it will take the same amout of bytes
+            if padding < 4:
+                    fmt += "c" * padding
```

Comment:
I would do it this way, since on no_dollars=True we also add another pointer to the payload.
```suggestion
            if padding < 4 + context.bytes * no_dollars:
                fmt += "c" * padding
```